docs/cep/windows/window-types/sliding-time-length.md

```diff
@@ -0,0 +1,33 @@
+---
+title: WINDOW SLIDING_TIME_LENGTH()
+---
+
+A _sliding time window_ that at a given time holds the last `length` events that arrived during last `time` period, and gets updated for every event arrival and expiration.
+
+## Syntax
+
+    WINDOW SLIDING_TIME_LENGTH(time <INT|LONG|TIME>, length <INT>)
+
+## Query Parameters
+
+| Name          | Description      | Default Value | Possible Data Types | Optional | Dynamic |
+|------------|-------------------------------------|---------------|---------------------|----------|---------|
+| time   | The sliding time period for which the window should hold events.   |               | INT LONG TIME       | No       | No      |
+| length | The number of events that should be be included in a sliding length window. |      | INT                 | No       | No      |
+
+## Example
+
+```sql
+CREATE STREAM cseEventStream (symbol string, price float, volume int);
+CREATE WINDOW cseEventWindow (symbol string, price float, volume int) SLIDING_TIME_LENGTH(2 sec, 10);
+
+@info(name = 'query0')
+INSERT INTO cseEventWindow
+FROM cseEventStream;
+
+@info(name = 'query1')
+INSERT all events INTO outputStream
+FROM cseEventWindow SELECT symbol, price, volume;
```

Comment:
@koshy-thomas or @StoyanTc, this code seems to be invalid at the SELECT according to validation. How do we fix it?

Reply:
The SELECT seems to be after the FROM here it should be before FROM..
`CREATE STREAM cseEventStream (symbol string, price float, volume int);
CREATE WINDOW cseEventWindow (symbol string, price float, volume int) SLIDING_TIME_LENGTH(2 sec, 10);
CREATE SINK outputStream WITH (type='stream', stream='outputStream') (symbol string, price float, volume int);

@info(name = 'query0')
INSERT INTO cseEventWindow
FROM cseEventStream;

@info(name = 'query1')
INSERT all events INTO outputStream
SELECT symbol, price, volume
FROM cseEventWindow;`

And I see outputStream is not defined in a lot of other examples too, so I am not sure how would that work? Maybe @StoyanTc can clarify...

Reply:
I believe that if the output stream is not defined that the GDN is smart enough to automagically create it. At any rate, the system says the code is valid.

Reply:
But how can the user access this outputStream is my question without defining it. I tried the without outputStream definition SW and I don't see the stream created for me in the console.. even if it's some kind of internal stream.. it still doesn't make sense.. since user should be able to access this stream via console I guess...

Reply:
Weird, because I don't think invisible streams are a thing. @StoyanTc, can you clarify?